src/compositions/useConnector.js

```diff
@@ -0,0 +1,48 @@
+import {
+  inject,
+  ref,
+  isRef,
+  watch,
+  onMounted,
+  onUnmounted,
+} from '../util/vue-compat';
+
+export function useConnector(connector, widgetParams, initialState) {
+  const instantSearchInstance = inject('$_ais_instantSearchInstance');
+  const getParentIndex = inject('$_ais_getParentIndex', undefined);
+  const indexWidget = getParentIndex && getParentIndex();
+  const parent = indexWidget || instantSearchInstance;
+  const state = ref(initialState);
```

Comment:
this is different to the widget mixin, I think it's not a bad idea per se, but not sure if we should introduce that parameter now or later

Reply:
Agree. I'm not sure if this is the best. It'd be good if we somehow can get initial state from each connector, so that we could use it here. Otherwise, widget's init happens in the next tick, meaning `useXYZ()` will have to return null for the first time. And it seems troublesome for users to always check if it's null.

Probably we could do something like

```js
// connectXYZ.js

export const initialState = {

}

export default function connectXYZ() { ... }
```

That way we get to keep the current shape of connectors, and also get the initial state without having to duplicate code over vue and react instant search.

Reply:
I actually thought a little more about it, and I think it makes more sense to have the state as null initially (like the mixin) to make integration for each connector easier (we don't even have to ship those initially)